src/cpp/session/modules/SessionRmdNotebook.R

```
@@ -357,14 +367,29 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
    if (is.null(outputFile))
       outputFile <- .rs.withChangedExtension(inputFile, ext = ".nb.html")
 
-   # TODO: pass encoding from frontend
-   encoding <- "UTF-8"
+   # attempt to get encoding from source database (note: this will only
+   # succeed for files already open in the IDE, but since this operation
+   # is normally called when attempting to preview / create a notebook on
+   # save we generally expect the document to be available)
+   properties <- .rs.getSourceDocumentProperties(inputFile)
+   
+   # attempt to read and re-encode the file to UTF-8 if it's specified
+   # in the system encoding
+   if (!identical(properties$encoding, "UTF-8")) {
```

there are a number of codepaths where `properties` can be null here (e.g. if the file does not exist or is not in the source database), so we should check for it (or maybe `.rs.getSourceDocProperties` should throw an error in those cases?)